arsenal/adapters/resource_type_yaml_parser.py

```diff
@@ -0,0 +1,29 @@
+# -*- coding: utf-8 -*-
+
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+import yaml
+
+
+class ResourceTypeYamlParser(object):
+
+    def __init__(self, resource_type_factory):
+        self.resource_type_factory = resource_type_factory
+
+    def fill_factory_from(self, filename):
```

Comment:
nit: semantics again... not so sure about fill_factory_from. i would suggest register_type or something like that